src/GridFieldActionsMenu.php

```diff
@@ -0,0 +1,315 @@
+    {
+        if ($columnName === 'Actions') {
+            return [
+                'title' => _t(__CLASS__ . '.MoreActions', 'More Actions'),
+            ];
+        }
+        return [];
+    }
+
+    public function getURLHandlers($gridField)
+    {
+        return [
+            'item/$ID//publish' => 'handleRecordAction',
+            'item/$ID//unpublish' => 'handleRecordAction',
+            'item/$ID//archive' => 'handleRecordAction',
+            'item/$ID' => 'handleRecordLink',
+        ];
+    }
+
+    /**
+     * Basically an overly condensed GridFieldDetailForm::handleItem
+     *
+     * @param GridField $gridField
+     * @param HTTPRequest $request
+     */
+    public function handleRecordLink($gridField, $request)
+    {
+        $injector = Injector::inst();
+        $requestHandler = $gridField->getForm()->getController();
+        $record = $gridField->getList()->byID($request->param("ID")) ?: $injector->create($gridField->getModelClass());
+        $handler = $injector->createWithArgs(
+            GridFieldDetailForm_ItemRequest::class,
+            [$gridField, $this, $record, $requestHandler, 'GridFieldActionsMenu']
+        );
+        return $handler->handleRequest($request);
+    }
+
+    /**
+     * Handle actions that don't require loading of a new page/panel/etc.
+     *
+     * @param GridField $gridField
+     * @param HTTPRequest $request
+     */
+    public function handleRecordAction($gridField, $request)
+    {
+        $record = $gridField->getList()->byID($request->param("ID"));
+        return GridFieldRecordActionHandler::create($gridField, $record)->handleRequest($request);
+    }
+
+    /**
+     * Set whether to include the first Root tab in the actions list
+     *
+     * @param bool $showFirstTab
+     * @return $this
+     */
+    public function setShowFirstTab($showFirstTab)
+    {
+        $this->showFirstTab = (bool) $showFirstTab;
+        return $this;
+    }
+
+    /**
+     * Get whether to include the first Root tab in the actions list
+     *
+     * @return bool
+     */
+    public function getShowFirstTab()
+    {
+        return $this->showFirstTab;
+    }
+
+    /**
+     * Set the actions to use in the dropdown menu
+     *
+     * @param array $actions
+     * @return $this
+     */
+    public function setActions(array $actions)
+    {
+        $this->actions = $actions;
+        return $this;
+    }
+
+    public function addActionToGroup(array $action, $groupName)
+    {
+        if (empty($this->actions[$groupName])) {
+            $this->actions[$groupName] = [];
+        }
+
+        $this->actions[$groupName][] = $action;
+
+        return $this;
+    }
+
+    /**
+     * Get the actions to use in the dropdown menu
+     *
+     * @return array
+     */
+    public function getActions()
+    {
+        return $this->actions;
+    }
+
+    // The following 3 null return functions implement an undefined interface
+    // expected by GridFieldDetailForm_ItemRequest
```

Comment:
This seems to be a symptom of mis-use of the API; You should pass in a GridFieldDetailForm to the GridFieldDetailForm_ItemRequest constructor, not an instance of this object.

Can you get the actual GridFieldDetailForm instance, and pass this in instead, in the code earlier in this file?